config/e3sm/config_grids.xml

```diff
@@ -1266,934 +1615,939 @@
-      <file lnd_mask="tx0.1v2">domain.lnd.conusx4v1_tx0.1v2.161129.nc</file>
-      <file ocn_mask="tx0.1v2">domain.ocn.conusx4v1_tx0.1v2.161129.nc</file>
     </domain>
 
-    <!-- Old mask for CONUS grid -->
     <domain name="tx0.1v2">
-      <nx>3600</nx>  <ny>2400</ny>
-      <file atm_mask="tx0.1v2">domain.ocn.tx0.1v2.090218.nc</file>
-      <file ice_mask="tx0.1v2">domain.ocn.tx0.1v2.090218.nc</file>
-      <file lnd_mask="tx0.1v2">domain.ocn.tx0.1v2.090218.nc</file>
-      <file ocn_mask="tx0.1v2">domain.ocn.tx0.1v2.090218.nc</file>
+      <nx>3600</nx>
+      <ny>2400</ny>
+      <file grid="atm|ice|lnd|ocn">$DIN_LOC_ROOT/share/domains/domain.ocn.tx0.1v2.090218.nc</file>
       <desc>tx0.1v2 is an old mask used for CONUS:</desc>
     </domain>
 
-
     <domain name="oQU240">
       <nx>7153</nx>
       <ny>1</ny>
+      <file grid="ice|ocn">$DIN_LOC_ROOT/share/domains/domain.ocn.oQU240.151209.nc</file>
       <desc>oQU240 is an MPAS ocean mesh with quasi-uniform 240 km grid cells, nominally 2 degree resolution:</desc>
-      <file ice_mask="oQU240">domain.ocn.oQU240.151209.nc</file>
-      <file ocn_mask="oQU240">domain.ocn.oQU240.151209.nc</file>
     </domain>
 
     <domain name="oQU240wLI">
       <nx>7268</nx>
       <ny>1</ny>
+      <file grid="ice|ocn">$DIN_LOC_ROOT/share/domains/domain.ocn.oQU240wLI.160929.nc</file>
       <desc>oQU240wLI is an MPAS ocean mesh with quasi-uniform 240 km grid cells, nominally 2 degree resolution. Additionally, it has ocean under landice cavities.:</desc>
-      <file ice_mask="oQU240wLI">domain.ocn.oQU240wLI.160929.nc</file>
-      <file ocn_mask="oQU240wLI">domain.ocn.oQU240wLI.160929.nc</file>
     </domain>
 
     <domain name="oQU120">
       <nx>28571</nx>
       <ny>1</ny>
+      <file grid="ice|ocn">$DIN_LOC_ROOT/share/domains/domain.ocn.oQU120.160401.nc</file>
       <desc>oQU120 is an MPAS ocean mesh with quasi-uniform 120 km grid cells, nominally 1 degree resolution:</desc>
-      <file ice_mask="oQU120">domain.ocn.oQU120.160401.nc</file>
-      <file ocn_mask="oQU120">domain.ocn.oQU120.160401.nc</file>
     </domain>
 
     <domain name="oRRS15to5">
       <nx>5778136</nx>
       <ny>1</ny>
+      <file grid="ice|ocn">$DIN_LOC_ROOT/share/domains/domain.ocn.oRRS15to5.160207.nc</file>
       <desc>oRRS15to5 is an MPAS ocean grid with a mesh density function that is roughly proportional to the Rossby radius of deformation, with 15 km gridcells at low and 5 km gridcells at high latitudes:</desc>
-      <file ice_mask="oRRS15to5">domain.ocn.oRRS15to5.160207.nc</file>
-      <file ocn_mask="oRRS15to5">domain.ocn.oRRS15to5.160207.nc</file>
     </domain>
 
   </domains>
 
-  <!-- ======================================================== -->
-  <!-- Mapping -->
-  <!-- ======================================================== -->
+  <!-- The following are the required grid maps that must not be idmap if the   -->
+  <!-- attributes grid1 and grid2 are not equal -->
+
+  <required_gridmaps>
```

Comment:
required_gridmaps was copied verbatim from cesm/config_grids.xml

Reply:
Required_gridmaps calls out the grid maps that will be required for interactions between grid1 and grid2.  There are three possible map types, flux, vector, scalar.  All three grids may not be required for interactions between a given pair of components.  I don't expect that there would be any difference here between cesm and e3sm.